### PR TITLE
Remove unnecessary multiplication and division for sprite lists

### PR DESCRIPTION
**Pull request by ZehMatt**
We only need to do that during s6 import/export now. I think this made the whole thing look more complex than it is

### CODE REVIEW

**Comment by Gymnasiast** [comment]
Looks OK. One thing though: * 2 is just multiplying by two, not taking things to the power of two.

**Comment by ZehMatt** [comment]
Yeah true I'll rename it

File: src/openrct2/world/Sprite.cpp [hunk 1]
```diff
@@ -299,7 +299,7 @@ static void sprite_reset(rct_sprite_generic* sprite)
 
     std::memset(sprite, 0, sizeof(rct_sprite));
 
-    sprite->linked_list_type_offset = llto;
+    sprite->linked_list_index = llto;
```
Comment (Gymnasiast) [inline src/openrct2/world/Sprite.cpp]: Perhaps change the name of the `llto` variable to `lli` accordingly.

File: src/openrct2/world/Sprite.cpp [hunk 2]
```diff
@@ -392,15 +392,13 @@ rct_sprite* create_sprite(uint8_t bl)
  * of the desired linked list in a uint16_t array. Known valid values are
  * 2, 4, 6, 8 or 10 (SPRITE_LIST_... * 2)
```
Comment (Gymnasiast) [inline src/openrct2/world/Sprite.cpp]: Comment needs changing.

**Comment by ZehMatt** [comment]
Changed

**Comment by Gymnasiast** [comment]
Merge after 0.2.3 release.

File: src/openrct2/rct12/RCT12.h [hunk 3]
```diff
@@ -377,25 +377,25 @@ assert_struct_size(RCT12EightCarsCorruptElement15, 8);
...
-    int16_t sprite_top;              // 0x18
-    int16_t sprite_right;            // 0x1A
-    int16_t sprite_bottom;           // 0x1C
-    uint8_t sprite_direction;        // 0x1E
+    uint8_t sprite_identifier;      // 0x00
+    uint8_t type;                   // 0x01
+    uint16_t next_in_quadrant;      // 0x02
+    uint16_t next;                  // 0x04
+    uint16_t previous;              // 0x06
+    uint8_t linked_list_index;      // 0x08
```
Comment (duncanspumpkin) [inline src/openrct2/rct12/RCT12.h]: Technically this one should have stayed as offset. Then it would make sense why you can't just copy it across.

Comment (Gymnasiast) [reply src/openrct2/rct12/RCT12.h]: Agreed.

Comment (ZehMatt) [reply src/openrct2/rct12/RCT12.h]: Agreed, I'll change it later today.

Comment (duncanspumpkin) [reply src/openrct2/rct12/RCT12.h]: Changed it myself :)

Comment (ZehMatt) [reply src/openrct2/rct12/RCT12.h]: Ha thanks, I was just about to start.

**Pull request merged**